src/core/Akka.Streams/Implementation/Fusing/Ops.cs

```diff
@@ -3817,19 +3817,7 @@ public override void OnDownstreamFinish()
                 _completionCts.Cancel();
                 _completionCts.Dispose();
 
-                try
-                {
-                    _enumerator.DisposeAsync().GetAwaiter().GetResult();
```

Comment:
The right solution here is to not dispose the iterator - iteration should stop automatically as a result of the `CancellationToken` and the `DisposeAsync` method won't do anything in scenarios where we're still waiting on the `Task` anyway. GC should still clean this up afterwards.

The alternative, adding a bunch of accounting code to safely call `DisposeAsync`, is really infeasible since we can't pass the `ValueTask` around (because it's a `struct`) and only supports a single `await`-er - unless we want to wrap it in a `Task` each time.

None of our other stages which retain an `IEnumerator<T>` call `Dispose` on it, so I don't think it's an issue.